### ISSUE TITLE
feature request: swagger integration

### ISSUE THREAD

**Issue by evist0** [logto-io/logto#3416]
### What problem did you meet?
I've been developing a simple backend that needs to integrate with Logto to validate the access token and so on.
My API needs to be tested somehow during development, so I connected the Swagger and tried to set up oidc authorization.

However, it did not work out for me due to Logto doesn't support `response_type=token` and `audience` param.
It can issue resource api tokens only via `token` endpoint with `resource` param.

### Describe what you'd like Logto to have
I want to be able to get resource api access tokens after going through the authorization flow.

1. Support `response_type=token`.
2. Add an `audience` param which indicates target resource api.

[Example Auth0 specification](https://auth0.com/docs/api/authentication#implicit-flow)

**Comment by gao-sun** [comment]
@simeng-li @wangsijie any ideas?

**Comment by simeng-li** [comment]
@evist0 The Implicit Authorization Flow you posted is strongly NOT recommended and has been deprecated by OAuth2.0. It may cause lots of security issues.  

Back to your requirements, have you tried our [M2M](https://docs.logto.io/docs/recipes/integrate-logto/machine-to-machine) solutions. It allows you to get authorized directly for your backend swagger service.

**Comment by evist0** [comment]
Swagger has support for OIDC (very truncated as it turned out) and OAuth 2, so I've tried to implement these two flows.
It also looked pretty natural: SPA App inside Logto -> Swagger initiates implicit auth flow -> Gets token.

I don't quite understand how to implement this under M2M, especially considering that the token should contain the user in the "sub" field, not the application.

**Comment by simeng-li** [comment]
Yes, M2M only grants authorizations to services, and the "sub" claim in the request token represents the app. 
Anyway, now I see, it sounds like a valid use case. 
Still, As a high-risk action this flow might be,  we need to carefully review the solutions before moving forward.  Whether this gonna be a development-only feature or open to production.  Letting admin users claim a revokable API secret key would be another way to achieve that. 
I'll keep you posted. Thx for your feedback, though.

**Comment by IceCoffee1024** [comment]
+1

**Comment by daniilium** [comment]
+1
